sorting/shell_sorting.cpp

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+	#include <vector>
+	#include <algorithm> 
+
+	using namespace std;
+
+	void shellSort(vector<int> &ar, int size)
+	{
+		int j;
+		for (int gap = size / 2; gap > 0; gap /= 2)
+	  	{
+	    	for (int i = gap; i < size; i++)
+	    	{
+	      		int temp = ar[i];
+	      		for (j = i; j >= gap && temp < ar[j - gap]; j -= gap)
+	      		{
+	        		ar[j] = ar[j - gap];
+	      		}
+	      		ar[j] = temp;
+	    	} 
+	  	}
+	}
+
+	int main()
```

Comment:
Please follow standard Cpp indent and spacing.

Please format the code accordingly.

Comments and documentation missing